### ISSUE TITLE
Remove shortcut from  Steam `Non-Steam Game` in the Steam Deck does not work sometimes

### ISSUE THREAD

**Issue by arielj** [Heroic-Games-Launcher/HeroicGamesLauncher#1905]
### Describe the bug

When I uninstall a game that was previously added as to Steam with the `Add to Steam` feature, the game does not get removed from Steam, even though the logs say it was.

EDIT: Using the `Remove from Steam` option also doesn't work, it does change the button to `Add to Steam` so it is doing something, but the game still shows up in the Steam library.

I think it might be related to me having 2 accounts registered? maybe adding is adding it to both accounts but removing is removing it from only one?

### Add logs

There's no error in the logs and it mentions a successful removal:

```
Remove Steam images from Black Widow: Recharged
Black Widow: Recharged was successfully removed from Steam
```

### Steps to reproduce

1. Install a game
2. Use the `Add to Steam` option
3. Restart in Game mode to see the game added to the library
4. Restart in Desktop mode and uninstall the game
5. Restart in Game mode to see the game is still in the library

### Expected behavior

Game should be removed from Steam when uninstalled.

### System Information

- OS [e. g. "Ubuntu"]: Steam Deck
- Heroic Version [e. g. 2.1.1]: 2.4.3

### Additional information

Not sure if it's relevant, but I have 2 accounts registered in the Steam Deck.

**Comment by arielj** [comment]
Adding some of my findings. The problem seems to be that I have 2 accounts AND that I changed to Game mode before removing the shortcut (which I don't understand why it's a factor in this issue).

So, this works:
- Add a shortcut (it gets added to both shortcuts.vdf)
- Remove the shortcut (it gets removed from both shortcuts.vdf)

This does NOT work:
- Add a shortcut (it gets added to both shortcuts.vdf)
- Restart in game mode (verify the game is showing up)
- Restart in desktop mode
- Remove the shortcut (it only gets removed on the shortucts.vdf for the account that was not logged in game mode :S)

**Comment by Nocccer** [comment]
Ok i see. I think there is a problem in my code. I will look later, but i think i just return as soon as i removed the game from one shortcuts.vdf. I need to iterate over all. Maybe thats the problem, because i don't care about steam accounts.

**Comment by flavioislima** [comment]
Is this still an issue?

**Comment by arielj** [comment]
yes, it's still happening for me (after removing steam shortcus, they are only removed from one steam account and I have to remove them manually inside steam). I'm not sure what's the fix though.

**Comment by Nocccer** [comment]
I saw this happens if you remove the game and steam is still open. Steam somehow caches the old shortcut.vdf and write it back if you close steam.

**Comment by arielj** [comment]
Closing this, I only had the issue with 2 steam accounts and haven't had 2 account in the same machine since then, it's either fixed or not important at this point.
